### PR TITLE
Use GitHub Actions to check for Dockerfile correctness

### PR DESCRIPTION
**Pull request by ariya**
This is to make sure that any future improvement to `Dockerfile` (such as PR #14694) will be tested in the pull request.

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    steps:
+    - uses: actions/checkout@v2
+    - name: Build container image
+      run: docker build -t test-container-image .
+    - name: Launch container
+      run: docker run -dp 3000:3000 test-container-image
+      timeout-minutes: 5
+    - run: docker ps
+    - name: Wait for Metabase to start
+      run: sleep 25
```
Comment (camsaul) [inline .github/workflows/docker.yml:23]: Are you sure Metabase will always start in 25 seconds? Maybe it's better to do something like what we do in CircleCI and wait for the port to start accepting incoming connections

Comment (ariya) [reply .github/workflows/docker.yml:23]: Good point. Somehow I couldn't make the `nc` check work (not sure why) but let me see if a loop with `curl` will do the trick.

Comment (ariya) [reply .github/workflows/docker.yml]: I've tweaked it so it doesn't wait for a hardcoded period of time anytime. Please take a look!

**Review by rlotun — APPROVED** [review]
lgtm

**Pull request merged**